src/asmcnc/comms/user_settings_manager.py

```diff
@@ -0,0 +1,127 @@
+     - a value: corresponding to their specific type
+
+     This settings manager provides getters and setters. Setters will do a validation check.
+
+     Create new settings:
+     Just add the new setting to the settings dictionary
+
+     Usage:
+
+     DON'T INSTANTIATE YOURSELF! GET IT FROM THE APP!
+     user_settings_manager = App.get_running_app().user_settings_manager
+
+     my_bool_attribute = user_settings_manager.get_value('dust_shoe_detection')
+     user_settings_manager.set_value('dust_shoe_detection', value)
+     my_title_label.text = user_settings_manager.get_title('dust_shoe_detection')
+     my_description_label.text = user_settings_manager.get_description('dust_shoe_detection')
+    """
+    SETTINGS_FILE_PATH = os.path.join(paths.SB_VALUES_PATH, "user_settings.json")
+
+    # local copy of settings (Add new settings here):
+    settings = {'dust_shoe_detection': {'description': 'When activated, the dust shoe needs to be inserted '
+                                                       'when starting the spindle or running jobs.',
+                                        'title': 'Dust shoe plug detection',
+                                        'type': 'bool',
+                                        'value': True},
+                'interrupt_bars_active': {'description': 'When activated, hitting the interrupt bars will trigger '
+                                                         'an alarm! Only deactivate this if your interrupt bar '
+                                                         'switches are broken!',
+                                          'title': 'Interrupt bars activated',
+                                          'type': 'bool',
+                                          'value': True}}
+
+    # dynamic property creation (kivy properties must be created at class level and not inside a class function:
+    for name, details in settings.items():
+        if details['type'] == 'bool':
+            vars()[name] = BooleanProperty(details['value'])
+        elif details['type'] == 'str':
+            vars()[name] = StringProperty(details['value'])
+        elif details['type'] == 'float':
+            vars()[name] = NumericProperty(details['value'])
+
+    def __init__(self, **kwargs):
+        Logger.info("Creating new instance of UserSettingsManager")
+        self.load_settings_from_file()
+        super(UserSettingsManager, self).__init__(**kwargs)
+
+    def load_settings_from_file(self):
+        """Loads the settings from the file and update the local dictionary."""
+        if os.path.exists(self.SETTINGS_FILE_PATH):
+            with open(self.SETTINGS_FILE_PATH, 'r') as f:
+                Logger.info('Loading settings from file: {}'.format(self.SETTINGS_FILE_PATH))
+                tmp = json.load(f)
+                self.settings.update(tmp)
+            # update properties
+            for name, details in self.settings.items():
+                if details['type'] == 'bool':
```

Comment:
why not store the type itself?

Reply:
tbh I didn't try that as I expected it gets mangled up when reading/writing the file. But I'll give it a try... (would also prefer that)

Reply:
As expected: TypeError: <type 'bool'> is not JSON serializable
I'll try to wrap in the load_ / save_ function....